vmdb/app/controllers/api_controller/providers.rb

```diff
@@ -0,0 +1,126 @@
+class ApiController
+  module Providers
+    CREDENTIAL_ATTRS = %w(userid password)
+    RESTRICTED_ATTRS = %w(type) + CREDENTIAL_ATTRS
+
+    def create_resource_providers(type, _id, data = {})
+      if data.key?("id") || data.key?("href")
+        raise BadRequestError,
```

Comment:
We could add ArgumentError to the api_controller/api_controller.rb ERROR_MAPPING.  If we do, we should go through all BadRequestError exceptions in the rest api code and identify which are ArgumentError and which are not and change all for consistency.